libpod/image/pull.go

```diff
@@ -254,6 +255,7 @@ func (ir *Runtime) doPullImage(ctx context.Context, sc *types.SystemContext, goa
 			io.WriteString(writer, fmt.Sprintf("Trying to pull %s...", imageInfo.image))
 		}
 		if err = cp.Image(ctx, policyContext, imageInfo.dstRef, imageInfo.srcRef, copyOptions); err != nil {
+			pullErrors = append(pullErrors, err)
```

Comment:
I feel like there are categories of errors we should ignore to reduce clutter (this is potentially going to produce downright gigantic error messages), but the differences in semantics between registry implementations may make this difficult to implement. For example, when implementing the debug logging, I noticed that I got three separate error messages for "no such image" from three different registries.

Reply:
> ...but the differences in semantics between registry implementations may make this difficult to implement.

Noisy and inconsistent seems better than short and very generic ;).  Can we land this and then massage any registry errors we want to consolidate later?

Reply:
As long as we drop all messages once successful, I am fine.

Reply:
> As long as we drop all messages once successful, I am fine.

We only consume this in the `if len(images) == 0` case below.